src/compare-internal.c

```diff
@@ -0,0 +1,38 @@
+#include <stdio.h>
+#include <vips/vips.h>
+
+
+const int STATS_SUM2_COLUMN = 3;
+
+/*
+  Compare images `actual` and `expected` and return mean square error.
+ */
+int Compare(VipsObject *context, VipsImage *actual, VipsImage *expected, double *out) {
+  if (actual->Bands != expected->Bands)
+    return -1;
+
+  if (actual->Type != expected->Type)
+    return -1;
+
+  VipsImage *difference;
+  VipsImage *absolute;
+  VipsImage *stats;
+  if (vips_subtract(actual, expected, &difference, NULL) ||
+      vips_abs(difference, &absolute, NULL) ||
+      vips_stats(absolute, &stats, NULL))
+    return -1;
+
+  vips_object_local(context, difference);
+  vips_object_local(context, absolute);
+  vips_object_local(context, stats);
+
+  double *statsData = (double*) stats->data;
+  int numValues = actual->Xsize * actual->Ysize * actual->Bands;
+  double sumOfSquares = statsData[STATS_SUM2_COLUMN];
+  double meanSquaredError = sumOfSquares / numValues;
```

Comment:
Based on VIPS source code for `vips_stats` and [Wikipedia](http://en.wikipedia.org/wiki/Mean_squared_error), but I am getting different values than ImageMagick’s `compare -metric mse`. Either way, it at least is 0 if the input images match.